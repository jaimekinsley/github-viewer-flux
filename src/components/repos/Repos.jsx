import React, { useEffect } from 'react';
import { useState } from '../../hooks/appContext';
import { fetchRepo } from '../../services/gitHubAPI';

const Repos = () => {
  const { state, dispatch } = useState();

  useEffect(() => {
    if(!state.user) return;
    fetchRepo(state.username)
      .then(res => {
        dispatch({
          type: 'SET_REPOS',
          payload: res
        });
      });
  }, [state.user]);

  const repoElements = state.repos.map(repo => (
    <li key={`${repo.html_url}+${repo.name}`}>
      <a href={repo.html_url}> {repo.name} </a>
    </li>
  ));

  return (
    <ul>
      {repoElements}
    </ul>
  );
};

export default Repos;
