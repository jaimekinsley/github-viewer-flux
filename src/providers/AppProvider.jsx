import React, { useReducer } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { useEffect } from 'react';
import { fetchUser } from '../services/gitHubAPI';

const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    fetchUser()
      .then(user => dispatch({ type: 'SET_USER', payload: user }));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppProvider;
