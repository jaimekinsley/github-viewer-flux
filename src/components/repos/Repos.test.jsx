import React from 'react';
import { mount } from 'enzyme';
import Repos from './Repos';
import AppProvider from '../../providers/AppProvider';

describe('Repos component', () => {
  it('renders Repos', () => {
    const wrapper = mount(<AppProvider><Repos /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
