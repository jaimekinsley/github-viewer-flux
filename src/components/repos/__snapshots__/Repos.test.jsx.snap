// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<AppProvider>
  <Repos>
    <ul />
  </Repos>
</AppProvider>
`;
